core/src/main/java/io/questdb/cairo/sql/SqlExecutionCircuitBreakerWrapper.java

```diff
@@ -38,13 +39,30 @@
 // However, the `delegate` circuit breaker instance referenced by the wrapper has to be thread-safe
 // if it is used by multiple threads (i.e. set as a delegate in multiple wrappers at the same time).
 public class SqlExecutionCircuitBreakerWrapper implements SqlExecutionCircuitBreaker, Closeable {
+    private final State state = new State();
+    private State backup;
     private SqlExecutionCircuitBreaker delegate;
     private NetworkSqlExecutionCircuitBreaker networkSqlExecutionCircuitBreaker;
 
     public SqlExecutionCircuitBreakerWrapper(@NotNull SqlExecutionCircuitBreakerConfiguration configuration) {
         networkSqlExecutionCircuitBreaker = new NetworkSqlExecutionCircuitBreaker(configuration, MemoryTag.NATIVE_CB2);
     }
 
+    public void backup() {
+        state.clear();
+        backup = null;
+        if (delegate != null) {
+            backup = state;
```

Comment:
backup behaviour is unorthodox, we assign state ref to backup, then change value in state ref, which changes both state and backup. At least some explanation would not go amiss

Reply:
Documented in 11abcb858ccbf6a31f3a80e56ec16f9275063d73